src/inja.hpp

```diff
@@ -574,11 +574,24 @@ class Renderer {
 						}
 						case Parsed::Loop::ForMapIn: {
 							const std::map<std::string, json> map = eval_expression<std::map<std::string, json>>(element_loop->list, data);
+                            unsigned int i = 0;
 							for (const auto& item: map) {
 								json data_loop = data;
-								data_loop[element_loop->key] = item.first;
-								data_loop[element_loop->value] = item.second;
-								result.append( render(Template(*element_loop), data_loop) );
+                                /* For nested loops, use parent/index */
+                                if (data_loop.count("index") == 1 && data_loop.count("index1") == 1) {
```

Comment:
Why does not to follow the jinja2 specs and not to put all these variables inside the 'loop' composite variable? In this case user can refer this vars as 'loop.index', 'loop.index1' etc. and can avoid situations like this:
`
{% for index in indexes %}
    {{ index }}
{% endfor %}
`

Reply:
Good question.  I'm not familiar with jinja2 - I just followed the existing construct that inja uses for loops over arrays.  However, your suggestion should be relatively straightforward to implement.

That said, I don't know how closely inja in intended to follow jinja2 specs - the inja readme says 'loosely inspired'.  I would defer to the opinion of @pantor in this case.

Reply:
Yep, I think we should follow the [jinja2 spec](http://jinja.pocoo.org/docs/2.10/templates/#list-of-control-structures) in this (and in general more) cases. I'll update the master branch to use the loop variable, and then feel free to update this PR, @craigpepper. Thanks so much!